akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMapAsyncSpec.scala

```diff
@@ -171,6 +171,16 @@ class FlowMapAsyncSpec extends AkkaSpec {
       c.expectComplete()
     }
 
+    "finish after future failure" in {
+      import system.dispatcher
+      Await.result(Source(1 to 3).mapAsync(1, n ⇒ Future {
```

Comment:
only when the last one to come in throws, actually (which made this test racy in the unordered case)